src/main/java/ch/uzh/ifi/hase/soprafs24/service/GameService.java

```diff
@@ -270,6 +279,26 @@ private void runActiveTurn(Game currentGame, Integer cardId) throws InterruptedE
         }
     }
 
+    public void handleInactivePlayer(Integer gameId) {
+        if (isUpdating.get()) return;
```

Comment:
is this to handle when running turns, this function is does not change anything?

Reply:
what happens when the turn timer has ended & the first card is picked?
imo, the turn timer would stop working, even though it is not the last card picking.

Reply:
this function is triggered through a rest endpoint that the frontend calls once the timer is up. isUpdating is required to block the inactive request in case the player click on a card before the turn is over, but the frontend has not yet received an answer. This is this race condition stuff.

Reply:
- when turn time ended and sent with REST request (inactive)
 - but the isUpdating is true because the activeTurn is running with 1st card
  => REST request falls in here(does nothing). and the active player does not change.